### PR TITLE
add orbit camera

### PR DESCRIPTION
**Pull request by rezural**
This loads camera type from config.ron, which defaults to Unreal,

and loads either the CameraPlugin, or the OrbitCameraPlugin.

Couple of possible issues:

I havent renamed CameraPlugin to UnrealCameraPlugin, which I can do.

See the FIXME: It's a bit ugly, but im not sure how else to have the OrbitController create the inner controller

### CODE REVIEW

File: src/camera.rs [hunk 1]
```diff
@@ -35,3 +55,34 @@ pub fn create_camera_entity(
...
+    //FIXME: This is ugly, but have lifetime/borrow issues if camera_controller holds a reference to inner_controller
+    let inner_controller = camera_controller.inner_controller();
+
+    commands
+        .spawn_bundle(PerspectiveCameraBundle {
+            transform: Transform::from_translation(eye).looking_at(target, Vec3::Y),
+            ..Default::default()
+        })
+        .insert(CursorRayCameraTag)
+        .insert(camera_controller)
```
Comment (bonsairobo) [inline src/camera.rs:85]: Is this necessary? I think you only have a query on `OrbitCamera`.

File: src/camera.rs [hunk 2]
```diff
@@ -35,3 +55,34 @@ pub fn create_camera_entity(
...
+            .add_event::<bevy_orbit_controls::CameraEvents>()
+            .add_plugin(bevy_orbit_controls::OrbitCameraPlugin)
+            .add_plugin(CursorRayPlugin);
+    }
+}
+
+pub fn create_orbit_camera_entity(commands: &mut Commands, eye: Vec3, target: Vec3) -> Entity {
+    let camera_controller = OrbitCameraController::new(eye, target, 10.0);
+    //FIXME: This is ugly, but have lifetime/borrow issues if camera_controller holds a reference to inner_controller
+    let inner_controller = camera_controller.inner_controller();
```
Comment (bonsairobo) [inline src/camera.rs:77]: I think you could just get rid of `OrbitCameraController`. It doesn't seem to buy you anything. Instead have a function that constructs the `OrbitCamera` how you want it.

**Review by bonsairobo — CHANGES_REQUESTED** [review]
Overall looking good, just a few changes I'd like.

**Comment by rezural** [comment]
OK, I'll update these in a bit.

Cheers

File: src/camera.rs (hunk 2 above)
Comment (rezural) [reply src/camera.rs]: Good point, fixed this.

File: src/camera.rs [hunk 3]
```diff
@@ -2,15 +2,35 @@ mod controller;
...
 use cursor_ray::CursorRayPlugin;
 
 use bevy::{
     app::prelude::*, ecs::prelude::*, math::prelude::*, render::prelude::*, transform::prelude::*,
 };
 
+pub fn create_camera_entity(
+    commands: &mut Commands,
+    control_config: CameraControlConfig,
+    config: Config,
```
[the remark this answers is not in the document]
Comment (rezural) [reply src/camera.rs:21]: I'll update this tomorrow. The rest of the patch is ready though.

**Pull request merged**